aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -210,25 +211,24 @@ void initialize(
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
     if (catalogProperties != null) {
       Map<String, String> tableSpecificCatalogProperties =
-          ImmutableMap.<String, String>builder()
-              .putAll(catalogProperties)
-              .put(
-                  AwsProperties.LAKE_FORMATION_DB_NAME,
-                  IcebergToGlueConverter.getDatabaseName(
-                      tableIdentifier, awsProperties.glueCatalogSkipNameValidation()))
-              .put(
-                  AwsProperties.LAKE_FORMATION_TABLE_NAME,
-                  IcebergToGlueConverter.getTableName(
-                      tableIdentifier, awsProperties.glueCatalogSkipNameValidation()))
-              .build();
+          Maps.newHashMapWithExpectedSize(catalogProperties.size() + 2);
+      tableSpecificCatalogProperties.putAll(catalogProperties);
+      tableSpecificCatalogProperties.put(
+          AwsProperties.LAKE_FORMATION_DB_NAME,
+          IcebergToGlueConverter.getDatabaseName(
+              tableIdentifier, awsProperties.glueCatalogSkipNameValidation()));
+      tableSpecificCatalogProperties.put(
+          AwsProperties.LAKE_FORMATION_TABLE_NAME,
+          IcebergToGlueConverter.getTableName(
+              tableIdentifier, awsProperties.glueCatalogSkipNameValidation()));
       // FileIO initialization depends on tableSpecificCatalogProperties, so a new FileIO is
       // initialized each time
       return new GlueTableOperations(
           glue,
           lockManager,
           catalogName,
           awsProperties,
-          initializeFileIO(tableSpecificCatalogProperties),
+          initializeFileIO(Collections.unmodifiableMap(tableSpecificCatalogProperties)),
```

Comment:
To test this, you could do `newTableOps` on a `GlueCatalog` and then ensure that the resulting `FileIO` can be round trip serialized through Kryo (which there's a helper method for I believe).

That way it won't be refactored back to `ImmutableMap`.

Reply:
+1, having a UT will certainly help us catching this, if this happens again. 

> To test this, you could do newTableOps on a GlueCatalog and then ensure that the resulting FileIO can be round trip serialized through Kryo (which there's a helper method for I believe).

+1 mostly did the same just wrote a new helper for it (not happy with it though, still thinking other alternatives), as at present Kryo helper depends on spark, in background includes Twitter#chill serializers. newTableOps presently have a protected access, so was facing issues in writing spark UT. 

The new helper takes a dependency in twitter-chill which spark uses and i think flink as well (may be wrong here, ref [link](https://flink.apache.org/news/2020/04/15/flink-serialization-tuning-vol-1.html#kryo)).

All suggestions are welcomed here :) !